### PR TITLE
build: use already present zombienet and polkadot binaries during tests [skip release]

### PR DESCRIPTION
**Pull request by 3h4x**
# Summary of changes

This PR speeds up setting up zombienet as it doesn't require fetching ~200MB binary each time it's run.
Additionally logs are grouped with time measurement and zombienet was updated to the latest.

Screenshot 2023-04-21 at 13 54 26

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [x] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [x] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [x] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### CODE REVIEW

File: tests/zombienet/zombienet.sh [hunk 1]
```diff
@@ -1,14 +1,13 @@
-# bash shebang
+#!/usr/bin/env bash
```
Comment (AwesomeIbex) [inline tests/zombienet/zombienet.sh:1]: this breaks nix users

File: tests/zombienet/zombienet.sh [hunk 2]
```diff
@@ -34,14 +33,21 @@ make_bin_dir() {
...
 fetch_zombienet() {
     if [ "$(nixos-version 2>/dev/null)" == "" ]; then
+        # Don't fetch zombienet if it's already present in the system
+        if which zombienet-$zombienet_version >/dev/null; then
+            cp $(which zombienet-$zombienet_version) "$bin_dir/zombienet"
+            echo "✅ Zombienet $zombienet_version installed"
+            return
+        fi
+
         if [ ! -f "$bin_dir/zombienet" ]; then
```
Comment (AwesomeIbex) [inline tests/zombienet/zombienet.sh:43]: We already do this here no?

File: tests/zombienet/zombienet.sh [hunk 3]
```diff
@@ -54,45 +60,52 @@ fetch_zombienet() {
...
     fi
 
     if [ ! -f "$polkadot_tmp_dir/$pdot_branch/target/release/polkadot" ]; then
         echo "Building polkadot..."
         cargo build --manifest-path "$polkadot_tmp_dir/$pdot_branch/Cargo.toml" --features fast-runtime --release --locked
-    else 
-        echo "Polkadot already built"
     fi
 
     if [ ! -f "$bin_dir/polkadot" ]; then
```
Comment (AwesomeIbex) [inline tests/zombienet/zombienet.sh:81]: this also already covers this check too

File: tests/zombienet/zombienet.sh (hunk 2 above)
Comment (3h4x) [reply tests/zombienet/zombienet.sh:43]: Not quite, each time this runs on a runner that havent fetched zombienet, or on a runner that $bin_dir was cleared we will have to refetch it. Github runner is usually creating some tmp dirs, which makes this action refetch zombienet each time.

What had changed in gh runner is that in /usr/bin/ we will have binaries like:
- zombienet-v1.4.40
- zombienet-v1.4.23
- polkadot-release-v0.9.37
- etc

So our script doesn't have to fetch repos, build anything as it's already present.

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**